tests/antithesis/Makefile

```diff
@@ -1,11 +1,24 @@
 REPOSITORY_ROOT := $(shell git rev-parse --show-toplevel)
 USER_ID := $(shell id -u)
 GROUP_ID := $(shell id -g)
+ARCH ?= $(shell go env GOARCH)
 
 .PHONY: antithesis-build-client-docker-image
 antithesis-build-client-docker-image:
 	docker build --build-arg GO_VERSION=$(shell cat $(REPOSITORY_ROOT)/.go-version) -f $(REPOSITORY_ROOT)/tests/antithesis/test-template/Dockerfile $(REPOSITORY_ROOT) -t etcd-client:latest
 
+.PHONY: antithesis-build-etcd-image
+antithesis-build-etcd-image:
+	git worktree add /tmp/etcd3.5 origin/release-3.5
```

Comment:
I'm not familiar with using git worktree this way. Can you explain what it's purpose? cc @ivanvc

Reply:
Sure.  To build the image from `release-3.5`, we need to check out the whole branch and can't retrieve just the dockerfile file as the binaries it needs have to be built from the code on the branch.  `git checkout origin/release-3.5` to build the image would fail when you have uncommitted changes, or if something fails during the build then git would still be on origin/release-3.5's commit.  So I think it's safer to do a temporary worktree and build there instead.

I don't think this is the best way to do it though.  I'm open to any suggestions or better ways to do it.

Reply:
I was not aware of this, if you look at https://github.com/etcd-io/etcd/blob/b605213c058c7ce4efda748ccb771ebee596b1af/tests/robustness/Makefile#L101-L197 my approach was very naive.

Reply:
got it.  do you prefer that way more?

Reply:
Don't know what would be the difference.

Reply:
technically, using worktree might be a bit faster, since it doesn't do another clone but uses the locally fetched commits.

Reply:
Make sense to me. Feel free to improve https://github.com/etcd-io/etcd/blob/b605213c058c7ce4efda748ccb771ebee596b1af/tests/robustness/Makefile#L101-L197 too